hamilton/plugins/h_pandas.py

```diff
@@ -144,52 +121,38 @@ def __init__(
+        :param select: The end nodes that represent columns to be appended to the original dataframe
+            via with_columns. Existing columns will be overridden.
         :param namespace: The namespace of the nodes, so they don't clash with the global namespace
             and so this can be reused. If its left out, there will be no namespace (in which case you'll want
             to be careful about repeating it/reusing the nodes in other parts of the DAG.)
         :param config_required: the list of config keys that are required to resolve any functions. Pass in None\
             if you want the functions/modules to have access to all possible config.
         """
 
-        self.subdag_functions = subdag.collect_functions(load_from)
-
-        if select is None:
-            raise ValueError("Please specify at least one column to append or update.")
-        else:
-            self.select = select
-
-        if (pass_dataframe_as is not None and columns_to_pass is not None) or (
-            pass_dataframe_as is None and columns_to_pass is None
-        ):
-            raise ValueError(
-                "You must specify only one of columns_to_pass and "
-                "pass_dataframe_as. "
-                "This is because specifying pass_dataframe_as injects into "
-                "the set of columns, allowing you to perform your own extraction"
-                "from the dataframe. We then execute all columns in the sbudag"
-                "in order, passing in that initial dataframe. If you want"
-                "to reference columns in your code, you'll have to specify "
-                "the set of initial columns, and allow the subdag decorator "
-                "to inject the dataframe through. The initial columns tell "
-                "us which parameters to take from that dataframe, so we can"
-                "feed the right data into the right columns."
-            )
-
-        self.initial_schema = columns_to_pass
-        self.dataframe_subdag_param = pass_dataframe_as
-        self.namespace = namespace
-        self.config_required = config_required
-
-    def required_config(self) -> List[str]:
-        return self.config_required
+        super().__init__(
+            *load_from,
+            columns_to_pass=columns_to_pass,
+            pass_dataframe_as=pass_dataframe_as,
+            select=select,
+            namespace=namespace,
+            config_required=config_required,
+            dataframe_type=DATAFRAME_TYPE,
+        )
 
     def _create_column_nodes(
-        self, inject_parameter: str, params: Dict[str, Type[Type]]
+        self, fn: Callable, inject_parameter: str, params: Dict[str, Type[Type]]
     ) -> List[node.Node]:
         output_type = params[inject_parameter]
 
-        def temp_fn(**kwargs) -> pd.DataFrame:
-            return kwargs[inject_parameter]
+        if inspect.iscoroutinefunction(fn):
+
+            async def temp_fn(**kwargs) -> Any:
```

Comment:
Curious why this is necessary -- we should be able to have it work with both sync and async? E.G. this function doesn't need to be awaited? Or is it that we end up awaiting this so we have to use an async function (even though its the same)? Comment would help.

Reply:
I just checked, it is not.. I thought since `extract_columns` is checking for a co-routine, we need to wire it through, but turns out it works without it as well. Same for the `create_merge_node`.